### ISSUE TITLE
Is the required parameter in function calling always included?

### ISSUE THREAD

**Issue by emorling** [minimaxir/simpleaichat#109]
Does simpleaichat populate the required parameter in function calling, so that all defined properties become required?

```
tools = [
    {
        "type": "function",
        "function": {
            "name": "get_current_weather",
            "description": "Get the current weather",
            "parameters": {
                "type": "object",
                "properties": {
                    "location": {
                        "type": "string",
                        "description": "The city and state, e.g. San Francisco, CA",
                    },
                    "format": {
                        "type": "string",
                        "enum": ["celsius", "fahrenheit"],
                        "description": "The temperature unit to use. Infer this from the users location.",
                    },
                },
                "required": ["location", "format"],  # <------------------------------- here
            },
        }
    }
]
```

**Comment by minimaxir** [comment]
tl;dr yes. All fields that aren't `Optional` or have a default value of `None` will set `required`.

This happens at the pydantic level, not simpleaichat, so you can test with your own schema and using `model_json_dump()`.
